distribution/lib/Standard/Base/0.0.0-dev/src/Data/Regression.enso

```diff
@@ -0,0 +1,108 @@
+
+    ## Fit a logarithmic line (y = A log x + B) to the data.
+    type Logarithmic_Model
+
+    ## Fit a power series (y = A x ^ B) to the data.
+    type Power_Model
+
+## Use Least Squares to fit a line to the data.
+fit_least_squares : Vector -> Vector -> Model -> Fitted_Model ! Illegal_Argument_Error | Fit_Error
+fit_least_squares known_xs known_ys model=Linear_Model =
+    Illegal_Argument_Error.handle_java_exception <| Fit_Error.handle_java_exception <| case model of
+        Linear_Model intercept ->
+            fitted = if intercept.is_nothing then Regression.fit_linear known_xs.to_array known_ys.to_array else
+                Regression.fit_linear known_xs.to_array known_ys.to_array intercept
+            Fitted_Linear_Model fitted.slope fitted.intercept fitted.rSquared
+        Exponential_Model intercept ->
+            log_ys = ln_series known_ys "Y-values"
+            fitted = if intercept.is_nothing then Regression.fit_linear known_xs.to_array log_ys.to_array else
+                Regression.fit_linear known_xs.to_array log_ys.to_array intercept.ln
+            fitted_model_with_r_squared Fitted_Exponential_Model fitted.intercept.exp fitted.slope known_xs known_ys
+        Logarithmic_Model ->
+            log_xs = ln_series known_xs "X-values"
+            fitted = Regression.fit_linear log_xs.to_array known_ys.to_array
+            fitted_model_with_r_squared Fitted_Logarithmic_Model fitted.slope fitted.intercept known_xs known_ys
+        Power_Model ->
+            log_xs = ln_series known_xs "X-values"
+            log_ys = ln_series known_ys "Y-values"
+            fitted = Regression.fit_linear log_xs.to_array log_ys.to_array
+            fitted_model_with_r_squared Fitted_Power_Model fitted.intercept.exp fitted.slope known_xs known_ys
+        _ -> Error.throw (Illegal_Argument_Error "Unsupported model.")
+
+type Fitted_Model
+    ## Fitted line (y = slope x + intercept).
+    type Fitted_Linear_Model slope:Number intercept:Number r_squared:Number=0.0
+
+    ## Fitted exponential line (y = a exp(b x)).
+    type Fitted_Exponential_Model a:Number b:Number r_squared:Number=0.0
+
+    ## Fitted logarithmic line (y = a log x + b).
+    type Fitted_Logarithmic_Model a:Number b:Number r_squared:Number=0.0
+
+    ## Fitted power series (y = a x ^ b).
+    type Fitted_Power_Model a:Number b:Number r_squared:Number=0.0
+
+    ## Display the fitted line.
+    to_text : Text
+    to_text =
+        equation = case self of
+            Fitted_Linear_Model slope intercept _ -> slope.to_text + " * X + " + intercept.to_text
+            Fitted_Exponential_Model a b _ -> a.to_text + " * (" + b.to_text + " * X).exp"
+            Fitted_Logarithmic_Model a b _ -> a.to_text + " * X.ln + " + b.to_text
+            Fitted_Power_Model a b _ -> a.to_text + " * X ^ " + b.to_text
+       "Fitted_Model(" + equation + ")"
+
+    ## Use the model to predict a value.
+    predict : Number -> Number
```

Comment:
I think this is fair for vectors, especially as they are not guaranteed to contain only numbers anyway. But for Table Columns I think we will need this to be vectorized in Java for optimal performance.

Reply:
If we get to doing this within the table then yes agree we will want to operate on a Vector (or perform the computation in Java).